src/lib.rs

```diff
@@ -350,32 +350,36 @@ fn locate_template_file(
     }
 }
 
-fn resolve_project_dir(name: &ProjectName, args: &Args) -> Result<PathBuf> {
-    if args.init {
-        let cwd = env::current_dir()?;
-        return Ok(cwd);
-    }
-
-    let dir_name = if args.force {
-        name.raw()
+fn resolve_project_dir(name: &ProjectName, args: &Args, dest: Option<&Path>) -> Result<PathBuf> {
+    if let Some(path) = dest {
```

Comment:
I suggest avoiding the big if / else branches by leveraging default unwrapping. 
Please have a look at da42a32aead7b55828fb0bf95b154474e800be25 where I refined the function a little to be more functional style and also more DRY than it was before.